File: 백준/Silver/11723. 집합/집합.py
import sys

if __name__ == '__main__':
    readline = sys.stdin.readline
    INIT_SET = 1
    for i in range(20):
        INIT_SET = (INIT_SET << 1) + 1
    M = int(readline())
    s = 0
    while M > 0:
        M -= 1
        cmd = readline().rstrip()
        if cmd == 'all':
            s=INIT_SET
        elif cmd == 'empty':
            s = 0
        else:
            cmd, n = cmd.split()
            x = 1 << (int(n) - 1)
            if cmd == 'add':
                s |= x
            elif cmd == 'remove':
                s &= ~x
            elif cmd == 'toggle':
                s ^= x
            elif cmd == 'check':
                print(1 if (s & x) else 0)

File: 백준/Silver/11723. 집합/집합.cc
#include <iostream>
#include <string>
using namespace std;

const int FULL_SET = (1 << 21) - 2;

int main() {
    ios::sync_with_stdio(0); cin.tie(0);
    int M, s = 0, n; 
    string cmd;

    cin >> M;
    while(M--) {
        cin >> cmd;
        if(cmd == "all") s = FULL_SET;
        else if(cmd == "empty") s = 0;
        else {
            cin >> n;
            if(cmd == "add") s |= 1 << n;
            else if(cmd == "remove") s &= ~(1 << n);
            else if(cmd == "check") cout << ((s & 1 << n) != 0) << '\n';
            else if(cmd == "toggle") s ^= 1 << n;
        }
    }
}